r1cs/gadgets/std/src/groups/curves/short_weierstrass/short_weierstrass_jacobian.rs

```diff
@@ -392,12 +661,14 @@ for AffineGadget<P, ConstraintF, F>
 
             //Perform addition
             let adder: Self = Self::new(x, y, Boolean::constant(false));
-            result = result.add(cs.ns(||format!("Add_{}", i)), &adder)?;
+            //TODO: Can we use add unsafe ?
```

Comment:
No we can't. Otherwise, a malicious prover may choose a crafted `shift`, which provocates exceptional cases in order to "exit" the curve. The random sampling of the shift is to allow an honest prover to efficiently find an execution path in which no exceptions are hit.